integrations/upgrade/index.test.ts

```diff
@@ -1467,6 +1467,139 @@ test(
   },
 )
 
+test(
+  'injecting `@config` in the shared root, when a tailwind.config.{js,ts,…} is detected',
+  {
+    fs: {
+      'package.json': json`
+        {
+          "dependencies": {
+            "@tailwindcss/upgrade": "workspace:^"
+          }
+        }
+      `,
+      'tailwind.config.ts': js`
+        export default {
+          content: ['./src/**/*.{html,js}'],
+          plugins: [
+            () => {
+              // custom stuff which is too complicated to migrate to CSS
+            },
+          ],
```

Comment:
```suggestion
          theme: {
            extend: {
              colors: {
                'my-red': 'red',
              },
            },
          },
```

When we update the config to something that we _can_ convert to, we noticed that the `@theme` now ends up in `src/index.css` and that the `@config` is still inside `src/tailwind-setup.css`.

Reply:
Added a failing test for this: https://github.com/tailwindlabs/tailwindcss/pull/14989/commits/a8a66a4aa62ad7d79bcc9a23ef9e104b47498f0e

Then fixed it: https://github.com/tailwindlabs/tailwindcss/pull/14989/commits/dc5cd8db12545c89852a2dae853d9193f537210f

Bonus points: basically deleted most of the code in the `migrate-config` file